_posts/2023-01-23-OpenSearch-Flink-Connector.markdown

```diff
@@ -0,0 +1,206 @@
+---
+layout: post
+title:  "Apache Flink® connector to OpenSearch®"
+authors:
+- ftisiot
+- reta
+date: 2022-01-31
+categories:
+ - technical-post
+
+excerpt: "Apache Flink® and OpenSearch® are widely known and successful open source projects. Even if you have never used them, it is very likely you have heard or read about them in the search or streaming data transformation use-cases. And there are reasons for that: although they are targeting different tech markets, they perform their job very well."
+---
+
+Apache Flink® and OpenSearch® are widely known and successful open source projects. Even if you have never used them, it is very likely you have heard or read about them in the search or streaming data transformation use cases. And there are reasons for that: although they are targeting different tech markets, they perform their job very well. 
+
+This post will help you understand how the Apache Flink Connector for OpenSearch bridges both projects together by enabling to push the outcome of data transformations directly to an OpenSearch index.
+
+![Flink + OpenSearch](flink_opensearch.png)
+
+## What is Flink?
+
+From the [official website](https://flink.apache.org/) of the project:
+
+> Apache Flink is a framework and distributed processing engine for stateful computations over unbounded and bounded data streams. Flink has been designed to run in all common cluster environments, perform computations at in-memory speed and at any scale.
+
+Apache Flink started from a fork of Stratosphere's distributed execution engine and became an Apache Incubator project in March 2014. In December 2014, Flink was accepted as an Apache top-level project.
+
+Flink is both a **framework**, providing the basics to create apps in Java, Python, and Scala, and a **distributed processing engine**, which can scale vertically and horizontally.
+It is used to create data pipelines over both unbounded (e.g. streaming) and bounded (batch) datasets, offering various levels of abstractions depending on the computation needs.
+Data is sourced and sinked via dedicated pluggable connectors, similarly to Apache Kafka. The Flink connector ecosystem is continuously evolving with new technologies being added regularly.
+
+All the principles listed above make Apache Flink a go-to open source data processing solution for a wide variety of industries and use cases.
+
+## Why did Aiven choose Flink over other options?
+
+Aiven applies dogfooding: Aiven runs on Aiven, meaning that we operate on the same data infrastructure we offer to our clients. Our standard approach is to evaluate open-source solutions for our internal needs, gain experience, make them robust and then, once we’re confident, offer them to our clients. 
+
+The same happened with Apache Flink, which was chosen because it’s open source (Aiven loves open source and we have a [dedicated OSPO team](https://aiven.io/blog/aivens-open-source-program-office)), and because it has a great and variegated community. 
```

Comment:
What about?

> We took the same approach with Flink, which was chosen because it’s open source (Aiven loves open source, and we have a [dedicated OSPO team](https://aiven.io/blog/aivens-open-source-program-office)) and because it has a large, varied and supportive community.

Reply:
@ftisiot Sounds great! Let's add the serial comma after "varied", though.

Reply:
@ftisiot could you please make it a suggestion so I would make the right change, thank you.

Reply:
```suggestion
We took the same approach with Flink, which was chosen because it’s open source (Aiven loves open source, and we have a [dedicated OSPO team](https://aiven.io/blog/aivens-open-source-program-office)) and because it has a large, varied, and supportive community.
```